opentutor_classifier/opentutor_classifier/lr/clustering_features.py

```diff
@@ -59,38 +60,62 @@ def alignment_metric(self, x, y) -> float:
             self.data[int(y[0])], self.data[int(x[0])]
         )
 
-    def fit_predict(self, data: np.ndarray):
+    def fit_predict(self, data: np.ndarray, train_quality):
         self.data = data
         x = np.arange(len(self.data)).reshape(-1, 1)
 
-        # Calculate pairwise distances with the new metric.
+        # Calculate pairwise distances with the new metric.'
         m = pairwise_distances(x, x, metric=self.alignment_metric)
 
         agg = AgglomerativeClustering(
-            n_clusters=2, affinity="precomputed", linkage="average"
+            n_clusters=train_quality, affinity="precomputed", linkage="average"
```

Comment:
direct pass through of `train_quality` here makes me nervous that someone will change the domain of train_quality elsewhere (e.g. 1-100) and then be confused by the effect.

Can we have limits, e.g.?

```python
CLUSTERS_MIN=1
CLUSTERS_MAX=2

n_clusters = min(CLUSTERS_MAX, max(CLUSTERS_MIN, train_quality))
...